server/sonar-server/src/main/java/org/sonar/ce/settings/ProjectConfigurationFactory.java

```diff
@@ -36,11 +40,28 @@ public ProjectConfigurationFactory(Settings globalSettings, DbClient dbClient) {
     this.dbClient = dbClient;
   }
 
-  public Configuration newProjectConfiguration(String projectKey) {
+  public Configuration newProjectConfiguration(String projectKey, Optional<Branch> branch) {
     Settings projectSettings = new ProjectSettings(globalSettings);
-    dbClient.propertiesDao()
-      .selectProjectProperties(projectKey)
-      .forEach(property -> projectSettings.setProperty(property.getKey(), property.getValue()));
+    addSettings(projectSettings, projectKey);
+    getBranchName(branch).ifPresent(
+      b -> addSettings(projectSettings, generateBranchKey(projectKey, b)));
     return new ConfigurationBridge(projectSettings);
   }
+
+  private void addSettings(Settings settings, String componentDbKey) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "Settings"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)